bathbot/src/commands/osu/rank/pp.rs

```diff
@@ -803,12 +803,25 @@ impl RankData {
                     pp_missing(user_pp, rank_holder_pp, scores)
                 };
 
+                let idx = idx + 1;
+
+                let idx_suffix = match idx % 100 {
+                    11 | 12 | 13 => "th",
+                    _ => match idx % 10 {
+                        1 => "st",
+                        2 => "nd",
+                        3 => "rd",
+                        _ => "th",
+                    },
+                };
+
                 format!(
                     "{prefix}, so {username} is missing **{missing}** raw pp, achievable \
-                    with a single score worth **{pp}pp** which would be the top #{idx}.",
+                    with a single score worth **{pp}pp** which would be their {idx}{suffix} top play.",
                     missing = WithComma::new(rank_holder_pp - user_pp),
                     pp = WithComma::new(required),
-                    idx = idx + 1,
+                    idx = idx,
+                    suffix = idx_suffix
```

Comment:
Instead of assigning variables their own name when formatting, one can just use the variable name directly so the `idx` line isn't necessary and the `suffix` line can be removed when renaming the `idx_suffix` variable to `suffix`.